packages/trader/src/Stores/Modules/Trading/Helpers/process.js

```diff
@@ -3,18 +3,29 @@ import * as ContractType from '../Actions/contract-type';
 import * as Duration from '../Actions/duration';
 import * as StartDate from '../Actions/start-date';
 
-export const processTradeParams = async (store, new_state) => {
+const processInSequence = async (store, functions) => {
     const snapshot = store.getSnapshot();
-    const functions = getMethodsList(store, new_state);
-
     // To make sure that every function is invoked and affects the snapshot respectively, we have to use for instead of forEach
     for (let i = 0; i < functions.length; i++) {
         // Shallow copy with Object.assign is good enough to extend the snapshot with new state
         // we don't need deep extension here, since each function in functions array composes a property of the store completely
         Object.assign(snapshot, await functions[i](snapshot)); // eslint-disable-line no-await-in-loop
```

Comment:
these `functions` are executed in sync bcoz the resulting state of the previous function should be passed to the next function